### PR TITLE
feat: disable eslint quotes, quoteProps, comma-spacing, and comma-dangle in typewriter

### PR DESCRIPTION
**Pull request by comcalvi**
Typewriter generates code that violates all of these. The eslint-only ones are harmless to disable by default, since we already disable `max-len` and these are also built-in with `eslint`. `comma-dangle` and `prettier/prettier` are not built-in, so we make these toggleable.

### CODE REVIEW

**Comment by mrgrain** [comment]
I'm not against this, but I'm surprised how this has only surfaced now?

File: packages/@cdklabs/typewriter/src/renderer/typescript.ts [hunk 1]
```diff
@@ -57,13 +57,29 @@ import { PrimitiveType, Type } from '../type';
...
+}
+
 export class TypeScriptRenderer extends Renderer {
+  private eslintPrettier: boolean;
+  private eslintCommaDangle: boolean;
+  public constructor(options: TypeScriptRenderOptions = {}) {
+    super(options);
+
+    this.eslintPrettier = options.eslintPrettier ?? false;
+    this.eslintCommaDangle = options.eslintCommaDangle ?? true;
```
Comment (mrgrain) [inline packages/@cdklabs/typewriter/src/renderer/typescript.ts:72]: If we do this, should we just go all in and take an array of eslint rules to disable?

**Comment by comcalvi** [comment]
> I'm not against this, but I'm surprised how this has only surfaced now?

I think it's because our `yargs` definition is one massive, 300-line expression.

File: packages/@cdklabs/typewriter/src/renderer/typescript.ts [hunk 2]
```diff
@@ -766,6 +786,12 @@ export class TypeScriptRenderer extends Renderer {
     }
   }
 
+  private renderEslint() {
+    this.emit(this.disabledEsLintRules.length > 0 ? '/* eslint-disable ' : '');
+    this.emitList(this.disabledEsLintRules, ', ', (rule) => this.emit(rule));
```
Comment (mrgrain) [inline packages/@cdklabs/typewriter/src/renderer/typescript.ts:791]:
```suggestion
    this.emitList(this.disabledEsLintRules, ',', (rule) => this.emit(rule));
```

**Review by mrgrain — APPROVED** [review]

**Pull request merged**